### ISSUE TITLE
Right to left text drawn in opposite order.

### ISSUE THREAD

**Issue by dhtdht020** [ruffle-rs/ruffle#3921]
**Describe the bug**
Right to left languages, like Hebrew, display incorrectly through Ruffle.
![image](https://user-images.githubusercontent.com/18469146/113918060-a4ebf280-97ea-11eb-8e78-cbe0750f216b.png)
Rather than drawing the letters from right to left, they are drawn in the opposite order.
So the following text (in hebrew): _Hello_
Appears as: _olleH_

**Expected behavior**
The text should be drawn from right to left.
In this specific example, the white text below the orange clock should be drawn as it does in Adobe Flash Player, to support flash animations that include RTL languages:

![image](https://user-images.githubusercontent.com/18469146/113917887-753cea80-97ea-11eb-9459-f667111bbfbd.png)

**Is the problem with the Ruffle desktop app, extension, or self-hosted version?**
I have tested this problem in both the Ruffle desktop app, and the (chrome) extension.

**What platform are you using?**
 - Windows 10, build 19042.
 - Latest nightly build as of writing (Nightly 2021-04-07)
 - Google Chrome, version 89.0.4389.114, 64bit.
 
 Link to SWF: [clock.zip](https://github.com/ruffle-rs/ruffle/files/6273851/clock.zip)

**Comment by pirzol** [comment]
same here.

**Comment by ousia** [comment]
Weird, using nightly from 2021-12-10 at https://ruffle.rs/demo/, I get only _x_.

![rtl-clock](https://user-images.githubusercontent.com/203767/145605421-f206673a-32b8-4cd0-91b4-d4c4e2e41a3c.png)

I wonder whether this may be a regression.

**Comment by ousia** [comment]
@ dhtdht020, how about opening a new issue with the regression displaying embedded font?

**Comment by dhtdht020** [comment]
> @dhtdht020, how about opening a new issue with the regression displaying embedded font?

Thanks, done that, #5839

**Comment by ousia** [comment]
@dhtdht020, the issue about displaying Hebrew characters has been fixed.

Could you confirm that Hebrew text isn’t displayed right to left (as it should be)?

Many thanks for your help.

**Comment by dhtdht020** [comment]
> @dhtdht020, the issue about displaying Hebrew characters has been fixed.
> 
> Could you confirm that Hebrew text isn’t displayed right to left (as it should be)?
> 
> Many thanks for your help.

Can confirm Hebrew text is not displayed right to left.

![image](https://user-images.githubusercontent.com/18469146/147688518-1e359f44-c858-4214-8f4b-9e0085d02be6.png)

**Comment by ousia** [comment]
@dhtdht020, is it just me or are Hebrew characters not being displayed again?

**Comment by dhtdht020** [comment]
> @dhtdht020, is it just me or are Hebrew characters not being displayed again?

Can confirm that it seems things have regressed again.

![image](https://user-images.githubusercontent.com/18469146/209016091-1f90bfe7-7877-4786-9b3a-f2d8ce1a41e1.png)

**Comment by ousia** [comment]
@dhtdht020, just reported the issue.

Many thanks for your confirmation.

**Comment by n0samu** [comment]
Right-to-left text is not rendered correctly in Ruffle at the moment, this is a known issue. If you see proper right-to-left text in any content, that would be because it is in the SWF as a bitmap/image rather than a text field.

**Comment by dhtdht020** [comment]
> @dhtdht020, after @n0samu stated that [this SWF doesn’t really use the glyphs from the embedded font](https://github.com/ruffle-rs/ruffle/issues/8885#issuecomment-1891294073), I wonder whether the lack of R2L is due to a flaw in _Ruffle_ or the provided SWF isn’t properly tagged.
> 
> Could you check whether the Hebrew version (or other RTL text) is working properly in https://htwins.net/scale2_old/?
> 
> Many thanks for your help.

I can confirm that the Hebrew version of the Flash app in the provided URL renders correctly and in the right order, with Nightly 2024-02-03.
